### ISSUE TITLE
Test sub removed with killed node does not pass constantly

### ISSUE THREAD

**Issue by vietj** [vert-x3/vertx-ignite#66]
The reproducer is to modify the `IgniteClusteredEventbusTest` with:

```
  @Rule
  public final RepeatRule rule = new RepeatRule();

  @Test
  @Repeat(times = 1000)
  @Override
  public void testSubsRemovedForKilledNode() throws Exception {
    super.testSubsRemovedForKilledNode();
  }
```

and run it

```
> mvn clean test -Dtest=IgniteClusteredEventbusTest#testSubsRemovedForKilledNode
```

the reported failure:

```
org.junit.ComparisonFailure: expected:<foo[2]> but was:<foo[1]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at io.vertx.test.core.AsyncTestBase.assertEquals(AsyncTestBase.java:350)
	at io.vertx.test.core.ClusteredEventBusTest.lambda$testSubsRemoved$15(ClusteredEventBusTest.java:231)
	at io.vertx.core.eventbus.impl.HandlerRegistration.deliver(HandlerRegistration.java:223)
	at io.vertx.core.eventbus.impl.HandlerRegistration.handle(HandlerRegistration.java:200)
	at io.vertx.core.eventbus.impl.EventBusImpl.lambda$deliverToHandler$3(EventBusImpl.java:533)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$2(ContextImpl.java:339)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:886)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by vietj** [comment]
can you check this @agura ?

**Comment by agura** [comment]
I'll take a look.

**Comment by zyclonite** [comment]
this was resolved latest with the cluster spi refactoring...

**Comment by tsegismont** [comment]
Closed in #82
